keylime/db/keylime_db.py

```diff
@@ -26,45 +27,65 @@ def make_engine(self, service):
-                                  config.get(service, 'database'))
-            # Create the path to where the sqlite database will be store with a perm umask of 077
-            os.umask(0o077)
-            kl_dir = os.path.dirname(os.path.abspath(database))
-            if not os.path.exists(kl_dir):
-                os.makedirs(kl_dir, 0o700)
-
-            url = URL(
-                drivername=drivername,
-                username=None,
-                password=None,
-                host=None,
-                database=(database)
-            )
-            engine = create_engine(url, connect_args={'check_same_thread': False},)
-        else:
-            url = URL(
-                drivername=drivername,
-                username=config.get(service, 'username'),
-                password=config.get(service, 'password'),
-                host=config.get(service, 'host'),
-                database=config.get(service, 'database')
-            )
-            engine = create_engine(url)
+        try :
+            p_sz_m_ovfl = config.get(service, 'database_pool_sz_ovfl')
+            p_sz, m_ovfl = p_sz_m_ovfl.split(',')
+        except NoOptionError :
+            p_sz = 5
+            m_ovfl = 10
 
-        return engine
+        engine_args = {}
+
+        url = config.get(service, 'database_url')
+        if url:
+            logger.info('database_url is set, using it for establishing database connection')
+            engine_args['pool_size'] = int(p_sz)
+            engine_args['max_overflow'] = int(m_ovfl)
+
+        else :
+            logger.info('database_url is not set, using multi-parameter database configuration options')
+
+            # This code shall be removed once we fully deprecate the old format
+            try :
+                drivername = config.get(service, "drivername")
+                logger.warning('Deprecation reminder: please add the suffix "database_" to all database-related parameters on your keylime.conf.')
+                p_n_prefix = ''
+            except NoOptionError :
+                drivername = config.get(service, 'database_drivername')
+                p_n_prefix = "database_"
 
+            if drivername == 'sqlite':
+                database = "%s/%s" % (config.WORK_DIR,
+                                    config.get(service, p_n_prefix + 'name'))
+                # Create the path to where the sqlite database will be store with a perm umask of 077
+                os.umask(0o077)
+                kl_dir = os.path.dirname(os.path.abspath(database))
+                if not os.path.exists(kl_dir):
+                    os.makedirs(kl_dir, 0o700)
+
+                url = URL(
+                    drivername=drivername,
+                    username=None,
+                    password=None,
+                    host=None,
+                    database=(database)
+                )
+                engine_args['connect_args'] = {'check_same_thread': False}
+
```

Comment:
Does this case not work with the new parameters?

Reply:
Yes. This is the one covered on the automated tests, actually (valid only for `sqlite`)